pkg/snapshot/snapshot.go

```diff
@@ -0,0 +1,90 @@
+package snapshot
+
+import (
+	"context"
+	"errors"
+
+	"github.com/acorn-io/baaah/pkg/router"
+	snapshotv1 "github.com/acorn-io/runtime/pkg/apis/snapshot.storage.k8s.io/v1"
+	"github.com/acorn-io/runtime/pkg/labels"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubectl/pkg/util/storage"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	ClassName = "acorn-snapshot-class"
+)
+
+// SyncSnapshotClasses creates the snapshot class using the default storage class.
+// May return an error.
+func SyncSnapshotClasses(req router.Request, resp router.Response) error {
+	// skip if no-object or deleted
+	if req.Object == nil || !req.Object.GetDeletionTimestamp().IsZero() {
+		return nil
+	}
+
+	storageClass := req.Object.(*storagev1.StorageClass)
+
+	// skip if not the default storage class
+	if storageClass.Annotations[storage.IsDefaultStorageClassAnnotation] != "true" {
+		return nil
+	}
+
+	resp.Objects(&snapshotv1.VolumeSnapshotClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: ClassName,
+			Labels: map[string]string{
+				labels.AcornManaged: "true",
+			},
+		},
+		Driver:         storageClass.Provisioner,
+		DeletionPolicy: "Delete",
+	})
+
+	return nil
+}
+
+// CreateSnapshotClass creates a snapshot class if one does not yet exist.
+// May return an error.
+func CreateSnapshotClass(ctx context.Context, client kclient.WithWatch) error {
+	acornSnapshotClass := &snapshotv1.VolumeSnapshotClass{}
+	err := client.Get(ctx, kclient.ObjectKey{
```

Comment:
I think the error should be handled if it is not nil.

Reply:
good point. fixed